apps/backoffice-v2/src/domains/entities/hooks/mutations/useRemoveDecisionTaskByIdMutation/useRemoveDecisionTaskByIdMutation.tsx

```diff
@@ -0,0 +1,71 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import toast from 'react-hot-toast';
+import { t } from 'i18next';
+import { TWorkflowById, updateWorkflowDecision } from '../../../../workflows/fetchers';
+import { workflowsQueryKeys } from '../../../../workflows/query-keys';
+import { useFilterId } from '../../../../../common/hooks/useFilterId/useFilterId';
+
+export const useRemoveDecisionTaskByIdMutation = (
+  workflowId: string,
+  postUpdateEventName?: string,
+) => {
+  const queryClient = useQueryClient();
+  const filterId = useFilterId();
+  const workflowById = workflowsQueryKeys.byId({ workflowId, filterId });
+
+  return useMutation({
+    mutationFn: ({ documentId }: { documentId: string }) =>
+      updateWorkflowDecision({
+        workflowId,
+        documentId,
+        body: {
+          decision: null,
+          reason: null,
+          postUpdateEventName,
+        },
+      }),
+    onMutate: async ({ documentId }) => {
+      await queryClient.cancelQueries({
+        queryKey: workflowById.queryKey,
+      });
+      const previousWorkflow = queryClient.getQueryData(workflowById.queryKey);
+
+      queryClient.setQueryData(workflowById.queryKey, (oldWorkflow: TWorkflowById) => {
+        return {
+          ...oldWorkflow,
+          context: {
+            ...oldWorkflow?.context,
+            documents: oldWorkflow?.context?.documents?.map(document => {
+              if (document?.id !== documentId) {
+                return document;
+              }
+
+              return {
+                ...document,
+                decision: {
+                  ...document?.decision,
+                  status: 'revision',
```

Comment:
Optimistic updates should reflect the expected response.
```suggestion
                  status: null,
```